os/arch/arm/src/common/up_schedyield.c

```diff
@@ -62,6 +66,18 @@ void up_schedyield(struct tcb_s *rtcb)
 		 * rtcb is the tcb of task which is yeilding the resource
 		 */
 		up_savestate(rtcb->xcp.regs);
+#ifdef CONFIG_ARMV8M_TRUSTZONE
+			/* Yes, then we should check if there is associated secure context
+			 * with the current active task. If yes, then store the secure context
+			 * before saving the non-secure context
+			 */
+			if (tz_memory) {
+				TZ_StoreContext_S(tz_memory);
+			}
+			rtcb->xcp.regs[REG_R8] = tz_memory;
```

Comment:
Is it okay to set the `rtcb->xcp.regs[REG_R8]` with `tz_memory` even if `tz_memory` is NULL?

Reply:
yes. while loading next task, REG_R8 in context register will be used to check if secure context is available or not. if REG_R8 is 0, it means no secure context.